Homework20

otatyanenko opened the pull request:
The issue of locators was solved.
I still sometimes have issue with deletePlaylist(), when the confirmation massage box sometimes does not appear.

ochaklosh committed:
Updated and Created

Updated
           ProfileTests
           build.gradle
Created Regression.xml

ochaklosh committed:
Created Homework18

Created
         SongTests.java
Added test playSong()
         Homework18.xml

ochaklosh committed:
Updated

Updated BaseTest by adding and updating existing (selectPlaylist()) helper method
Updated PlaylistTests, added flexibility by addressing the issue if the playlist does not exist and we are trying to delete it. As well as updated createPlaylist to address if the playlist already exists as well as to check if the user is already logged in

ochaklosh committed:
Updated

Updated helper methods in BaseTest by adding explicit waits
Currently working on PlaylistTest() on addSong() to allow the song to be added to new playlist

ochaklosh committed:
Updated

Finally updated locators for input and submit button that work for the
addNewPlaylist()
Other updates for waits and clean up

ochaklosh committed:
Cleaned for homework20

khanhtestpro commented:
//Comment from HW team,
Hello,
Your homework has been reviewed.
Great job on creating your Homework 20 and deletePlaylist() - implement Explicit waits. Regarding the ok button, it only appears when there is a song in the playlist, so please add the song to the playlist. Please check
https://share.zight.com/6qupWY65
Best wishes from the HW team (Khanh, khanh@testpro.io)

ochaklosh committed:
Update PlaylistTests.java

Updated deletePlaylist() added helper method checkIfPlaylistEmpty(), after home checker reminded me that the confirmation box appears only if the Playlist is not empty.

ochaklosh committed:
Cleaned up PlaylistTests

Moved helper method to BaseTest
